### PR TITLE
LISA-2403: Added feature flag to send an annual return API

### PR DESCRIPTION
**Pull request by chriswaugh**
It's now possible to disable this endpoint via config.

There is a "disabledEndpoints" list in the config, adding "annual-returns" to this will disable the endpoint and cause a NOT_IMPLEMENTED error to be returned to the user.

### CODE REVIEW

File: test/unit/controllers/APIVersioningSpec.scala [hunk 1]
```diff
@@ -85,20 +86,40 @@ class APIVersioningSpec extends PlaySpec with MockitoSugar with OneAppPerSuite {
...
   }
 
+  "The isEndpointEnabled function" must {
+    "allow execution if the appContext returns that the endpoint is enabled" in {
+      val request: FakeRequest[AnyContentAsEmpty.type] = FakeRequest(Helpers.GET, "/").withHeaders((ACCEPT, "application/vnd.hmrc.2.0+json"))
+
+      when(mockAppContext.endpointIsEnabled("test")).thenReturn(true)
+
+      isEndpointEnabledTest(request) mustBe Results.Ok
+
```
Comment (drewmiley) [inline test/unit/controllers/APIVersioningSpec.scala]: Remove line

File: app/uk/gov/hmrc/lisaapi/config/AppContext.scala [hunk 2]
```diff
@@ -40,4 +39,10 @@ class AppContext @Inject()(
   lazy val v2endpointsEnabled = runModeConfiguration.getBoolean("api.endpointsEnabledv2").getOrElse(throw new RuntimeException(s"Missing key api.endpointsEnabledv2"))
 
   override protected def mode = environment.mode
+
+  def endpointIsEnabled(endpoint: String): Boolean = {
+    val endpointIsDisabled = runModeConfiguration.getStringList("api.disabledEndpoints").fold(false)(list => list.contains(endpoint))
```
Comment (drewmiley) [inline app/uk/gov/hmrc/lisaapi/config/AppContext.scala]: Should the default be true or false?

File: app/uk/gov/hmrc/lisaapi/config/AppContext.scala [hunk 3]
```diff
@@ -40,4 +39,10 @@ class AppContext @Inject()(
...
+
+    !endpointIsDisabled
```
Comment (drewmiley) [inline app/uk/gov/hmrc/lisaapi/config/AppContext.scala]: One line it?

Comment (chriswaugh) [reply app/uk/gov/hmrc/lisaapi/config/AppContext.scala]: I'd considered it but the logic was a little hard to follow. Going to refactor a little to make it simpler...

File: app/uk/gov/hmrc/lisaapi/config/AppContext.scala (hunk 2 above)
Comment (chriswaugh) [reply app/uk/gov/hmrc/lisaapi/config/AppContext.scala]: I think the default state should be that every API is available unless they're explicitly disabled. Then if we have every API available there's no config required at all.

File: app/uk/gov/hmrc/lisaapi/config/AppContext.scala (hunk 3 above)
Comment (chriswaugh) [reply app/uk/gov/hmrc/lisaapi/config/AppContext.scala]: Refactored in dfc9ff6

File: app/uk/gov/hmrc/lisaapi/config/AppContext.scala (hunk 2 above)
Comment (drewmiley) [reply app/uk/gov/hmrc/lisaapi/config/AppContext.scala]: Could we use notContains? Or have the method as endpointIsDisabled?

Comment (drewmiley) [reply app/uk/gov/hmrc/lisaapi/config/AppContext.scala]: Hahaha, already done so!
